packages/contracts-bedrock/test/L2/SuperchainERC20.t.sol

```diff
@@ -0,0 +1,231 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.25;
+
+// Testing utilities
+import { Test } from "forge-std/Test.sol";
+
+// Libraries
+import { Predeploys } from "src/libraries/Predeploys.sol";
+import { IERC20 } from "@openzeppelin/contracts-v5/token/ERC20/IERC20.sol";
+import { IL2ToL2CrossDomainMessenger } from "src/L2/IL2ToL2CrossDomainMessenger.sol";
+
+// Target contract
+import {
+    CallerNotL2ToL2CrossDomainMessenger,
+    InvalidCrossDomainSender,
+    SuperchainERC20,
+    ISuperchainERC20Extensions,
+    ZeroAddress
+} from "src/L2/SuperchainERC20.sol";
+
+/// @notice Mock contract for the SuperchainERC20 contract so tests can mint tokens.
+contract SuperchainERC20Mock is SuperchainERC20 {
+    constructor(
+        string memory _name,
+        string memory _symbol,
+        uint8 _decimals
+    )
+        SuperchainERC20(_name, _symbol, _decimals)
+    { }
+
+    function mint(address _account, uint256 _amount) public {
+        _mint(_account, _amount);
+    }
+}
+
+/// @title SuperchainERC20Test
+/// @notice Contract for testing the SuperchainERC20 contract.
+contract SuperchainERC20Test is Test {
+    address internal constant ZERO_ADDRESS = address(0);
+    string internal constant NAME = "SuperchainERC20";
+    string internal constant SYMBOL = "SCE";
+    uint8 internal constant DECIMALS = 18;
+    address internal constant MESSENGER = Predeploys.L2_TO_L2_CROSS_DOMAIN_MESSENGER;
+
+    SuperchainERC20 public superchainERC20Impl;
+    SuperchainERC20Mock public superchainERC20;
+
+    /// @notice Sets up the test suite.
+    function setUp() public {
+        superchainERC20 = new SuperchainERC20Mock(NAME, SYMBOL, DECIMALS);
+    }
+
+    /// @notice Helper function to setup a mock and expect a call to it.
+    function _mockAndExpect(address _receiver, bytes memory _calldata, bytes memory _returned) internal {
+        vm.mockCall(_receiver, _calldata, _returned);
+        vm.expectCall(_receiver, _calldata);
+    }
+
+    /// @notice Test that the contract's `initializer` sets the correct values.
+    function test_initializer_succeeds() public view {
```

Comment:
should this be constructor instead of initializer or did you use the initializer word due to the unstructured storage?